samples/net/sockets/big_http_download/src/big_http_download.c

```diff
@@ -23,24 +23,42 @@
 #include <net/socket.h>
 #include <kernel.h>
 #include <net/net_app.h>
+#include <net/net_tls.h>
```

Comment:
Hmm, are you proposing that we have two almost identical samples, the other one with TLS support and the other one without?

Reply:
My goal was to wrap TLS related, non-posix changes with `CONFIG_NET_SOCKETS_SOCKOPT_TLS`, so that the samples in basic configuration will remain intact, and they still can be built with the Makefiles posix. I see that I've messed in `big_http_download` (there are errors, trivial but still), but `http_get` follows this pattern.

Reply:
> Hmm, are you proposing that we have two almost identical samples, the other one with TLS support and the other one without?

No, I'm proposing that the original requirement that the current samples are being POSIX samples, showing their portability to Zephyr - remains intact. If that requires "two almost identical samples", yes.